src/fvectorbin/nnet3-fvector-get-egs.cc

```diff
@@ -0,0 +1,211 @@
+// fvectorbin/nnet3-fvector-get-egs.cc
+
+// Copyright 2012-2016  Johns Hopkins University (author:  Daniel Povey)
+
+// See ../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#include <sstream>
+
+#include "util/common-utils.h"
+#include "nnet3/nnet-example.h"
+
+namespace kaldi {
+namespace nnet3 {
+
+// A struct for holding information about the position and
+// duration of each pair of chunks.
+struct FvectorChunkPairInfo {
+  std::string pair_name;
+  std::string utt_a;
+  std::string utt_b;
+  int32 output_archive_id;
+  int32 start_frame;
+  int32 num_frames;
+};
+
+// Process the range input file and store it as a map from utterance
+// name to vector of ChunkPairInfo structs.
+static void ProcessRangeFile(const std::string &range_rxfilename,
+                             std::vector<FvectorChunkPairInfo *> *pairs) {
+  Input range_input(range_rxfilename);
+  if (!range_rxfilename.empty()) {
+    std::string line;
+    while (std::getline(range_input.Stream(), line)) {
+      FvectorChunkPairInfo *pair = new FvectorChunkPairInfo();
+      std::vector<std::string> fields;
+      SplitStringToVector(line, " \t\n\r", true, &fields);
+      if (fields.size() != 6) {
+        KALDI_ERR << "Expected 6 fields in line of range file, got "
+                  << fields.size() << " instead.";
+      }
+
+      std::string utt_a = fields[0],
+                  utt_b = fields[1],
+                  start_frame_str = fields[4],
```

Comment:
fields[2] and fields[3]. You just copied from David's code without changing the name!!